cdap-ui/app/features/hydratorplusplus/services/create/stores/left-panel-store.js

```diff
@@ -81,30 +79,20 @@ class HydratorPlusPlusLeftPanelStore {
           this.state.defaultArtifactMap = res;
           return this.$q.resolve;
         })
-        .then(this.cleanupNonExistantPlugins.bind(this));
-
-    let dispatcher = HydratorPlusPlusLeftPanelDispatcher.getDispatcher();
-    dispatcher.register('onLeftPanelToggled', this.setState.bind(this));
-    dispatcher.register('toggleLeftPanelState', this.togglePanelState.bind(this));
+        .then($timeout(this.cleanupNonExistantPlugins.bind(this), 10000));
```

Comment:
So what ```cleanupNonExistantPlugins``` does is to check if some plugins have been removed, if yes then update the userstore if those plugins have a default version as that data is stale and needs to be cleanedup.

Previously we used to fetch only source, transform & sink and check if those three are fetched if not repeatedly wait (using `$timeout`) until they are fetched and then cleanup. 

What happened now is everything is dynamic (fetching extensions, fetching plugins under each extensions). So now the source-transform-sink trio is no longer valid (as there could be other extensions). 

What I should have done is check how many extension are there and check if all extensions' plugins are fetched and THEN cleanup. Since this is a little bit more involved work, considering the current time constraint I am defaulting it to 10s (hoping we get all plugins within 10s). 

Will definitely make this hack in the next PR to cleanup